src/clients/http.rs

```diff
@@ -22,15 +22,17 @@ use url::Url;
 
 use crate::health::{HealthCheckResult, HealthStatus, OptionalHealthCheckResponseBody};
 
+pub type ClientImpl = reqwest_middleware::ClientWithMiddleware;
+
 #[derive(Clone)]
 pub struct HttpClient {
     base_url: Url,
     health_url: Url,
-    client: reqwest::Client,
+    client: ClientImpl,
```

Comment:
The switch from `reqwest::Client` to `reqwest_middleware::Client` makes essentially 0 logical changes and hopefully isnt too intrusive of a change